src/docGenerator.ts

```diff
@@ -0,0 +1,146 @@
+import fs from "fs"
+import { JSONSchema4 } from "json-schema"
+import { flatMap, flatMapDeep, partition, toPairs } from "lodash"
+import fetch from "node-fetch"
+import { promisify } from "util"
+import { defaultEngine } from "./eslintDefaultOptions"
+import { DescriptionEntry } from "./model/Description"
+import {
+  Category,
+  fromEslintCategoryToCategory,
+  fromEslintCategoryToLevel,
+  Level,
+  patternIdToCodacy,
+  Patterns,
+  PatternsEntry,
+  PatternsParameter
+} from "./model/Patterns"
+import { toolName, toolVersion } from "./toolMetadata"
+
+const writeFile = promisify(fs.writeFile)
+
+export function generatePatterns(): Patterns {
+  const rules = defaultEngine.getRules()
+  const patterns = Array.from(rules.entries()).map(
+    ([patternId, ruleModule]) => {
+      const meta = ruleModule && ruleModule.meta ? ruleModule.meta : undefined
+      const eslintCategory = meta && meta.docs ? meta.docs.category : undefined
+      const level: Level = fromEslintCategoryToLevel(eslintCategory)
+      const category: Category = patternId.includes("security")
```

Comment:
are we flattening the security subcategories?

Reply:
Yes, I don't see an easy way to filter out the security subcategories.
I would leave this to a subsequent PR, if you are ok with it.